pkg/apis/operator/v1alpha1/tektonpipeline_defaults.go

```diff
@@ -106,7 +106,7 @@ func (p *Pipeline) setDefaults() {
 	}
 
 	if p.EnableStepActions == nil {
-		p.EnableStepActions = ptr.Bool(config.DefaultEnableStepActions.Enabled)
+		p.EnableStepActions = ptr.Bool(true)
```

Comment:
we need to handle the upgrade scenario too

Reply:
Good catch 
Updated now
PTAL thank you

Reply:
i dont understand how it will handle the upgrade scenario and will also different between what we are doing in `if` and `else`

for upgrade, we need to handle it in pre upgrade reconciler.

Reply:
> for upgrade, we need to handle it in pre upgrade reconciler.

May be i am missing something 
I know about pre and post reconcile part of the code
can you point me to some link how this pre upgrade reconciler works in Operator

Reply:
Done PTAL thank you